oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3DataStoreFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.jackrabbit.oak.blob.cloud.s3;
+
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.ConfigurationPolicy;
+import org.apache.jackrabbit.core.data.DataStore;
+import org.apache.jackrabbit.oak.plugins.blob.datastore.AbstractDataStoreFactory;
+import org.osgi.service.component.ComponentContext;
+
+import java.util.Map;
+
+@Component(policy = ConfigurationPolicy.REQUIRE,
+        name = "org.apache.jackrabbit.oak.plugins.blob.datastore.S3DataStoreFactory",
+        configurationFactory = true,
+        label = "Apache Jackrabbit Oak S3DataStore Factory",
+        description = "Factory allowing configuration of multiple instances of S3DataStores, " +
+                "for use with CompositeDataStore.")
+public class S3DataStoreFactory extends AbstractDataStoreFactory {
+    @Override
+    protected DataStore createDataStore(ComponentContext context, Map<String, Object> config) {
+        DataStore ds = S3DataStoreService.createS3DataStore(context, config, getStatisticsProvider(), getDescription(), closer);
```

Comment:
Important piece missing is the init of the DataStore. Calling just the createS3DataStore()/createDataStore() and missing parts of the registerDataStore() in the AbstractDataStoreService would fail to initialize the DataStore itself.
Not sure you have tested with the S3DataStore delegate but I don't think that works at present.

Reply:
Ok that's why I missed it but what is still not being done is the injection of properties via the PropertyUtils. This is needed for some props for the DataStore including the caching layer.
It would be better to initialize them (properties injection and calling init()) in the service or factory.